File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void sort012(vector<int>& arr) {
    int zerocount = 0;
    int onecount = 0;
    int twocount = 0;

  
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] == 0) {  
            zerocount++;
        } else if (arr[i] == 1) {
            onecount++;
        } else if (arr[i] == 2) {
            twocount++;
        }
    }

    int index = 0;


    while (zerocount--) {
        arr[index] = 0;  
        index++;
    }


    while (onecount--) {
        arr[index] = 1;  
        index++;
    }


    while (twocount--) {
        arr[index] = 2;  
        index++;
    }
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Medium/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.py
#{ 
 # Driver Code Starts

# } Driver Code Ends
class Solution:
    # Function to sort an array of 0s, 1s, and 2s
    def sort012(self, arr):
        countone = 0
        countzero = 0
        counttwo = 0
        
        n = len(arr)
        for i in range(n):
            if arr[i] == 0:
                countzero += 1
            if arr[i] == 1:
                countone += 1
            if arr[i] == 2:
                counttwo += 1
        
        index = 0
        
        while countzero > 0:
            
            arr[index] = 0
            
            index +=1
            countzero -=1
            
        while countone > 0:
            
            arr[index] = 1
            
            index +=1
            countone -=1
            
        while counttwo > 0:
            
            arr[index] = 2
            
            index +=1
            counttwo -=1  
            
            
            

#{ 
 # Driver Code Starts.
def main():
    t = int(input().strip())  # Read the number of test cases
    ob = Solution()

    while t > 0:
        t -= 1
        arr = list(map(int,
                       input().strip().split())
                   )  # Read the array as space-separated integers
        ob.sort012(arr)  # Sort the array

        print(' '.join(map(str, arr)))  # Print the sorted array
        print("~")
        
if __name__ == "__main__":
    main()

# } Driver Code Ends